lte/gateway/python/magma/pipelined/encoding.py

```diff
@@ -13,24 +13,125 @@
 
 import codecs
 import hashlib
+import gzip
+from Crypto.Cipher import ARC4
+from Crypto.Cipher import AES
+from Crypto.Hash import HMAC
+from Crypto.Random import get_random_bytes
 from lte.protos.mconfig.mconfigs_pb2 import PipelineD
 
 
-def encrypt_str(s: str, *_):
-    return s
+def pad(m):
+    return m + ' ' * (16 - len(m) % 16)
 
 
-def get_hash(s: str, hash_function):
-    hash_hex = ""
+def encrypt_str(s: str, key: bytes, encryption_algorithm, mac: bytes = None):
+    ret = ""
+    if encryption_algorithm == PipelineD.HEConfig.RC4:
+        cipher = ARC4.new(key)
+        ret = cipher.encrypt(s).hex()
+    elif encryption_algorithm == PipelineD.HEConfig.AES256_CBC_HMAC_MD5:
+        iv = get_random_bytes(16)
+        key_val = key
+        key_mac = mac
+
+        cipher = AES.new(key_val, AES.MODE_CBC, iv)
+        enc = cipher.encrypt(pad(s))
+
+        hmac = HMAC.new(key_mac)
+        hmac.update(iv + enc)
+
+        ret = hmac.hexdigest() + iv.hex() + enc.hex()
+    elif encryption_algorithm == PipelineD.HEConfig.AES256_ECB_HMAC_MD5:
+        key_val = key
+        key_mac = mac
+
+        cipher = AES.new(key_val, AES.MODE_ECB)
+        enc = cipher.encrypt(pad(s))
+
+        hmac = HMAC.new(key_mac)
+        hmac.update(enc)
+
+        ret = hmac.hexdigest() + enc.hex()
+    elif encryption_algorithm == PipelineD.HEConfig.GZIPPED_AES256_ECB_SHA1:
+        key_val = key
+        key_mac = mac
+
+        cipher = AES.new(key_val, AES.MODE_ECB)
+        enc = cipher.encrypt(pad(s))
+
+        hmac = HMAC.new(key_mac)
+        hmac.update(enc)
+        ret = gzip.compress(hmac.digest() + enc)
```

Comment:
For forward-looking bug safety: throw an exception (or just log the err, up to you) as the catch-all else

This handles the case where a new config option is added, but not accounted for here -- in which case an empty string would be returned, which is harder to debug/track down than an immediate exception